### PR TITLE
Add cosine_similarity(array(double), array(double))

### PR DESCRIPTION
**Pull request by dain**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## General
* Add `cosine_similarity(array(double), array(double))`. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/scalar/ArrayVectorFunctions.java [hunk 1]
```diff
@@ -56,6 +56,14 @@ public static double dotProduct(@SqlType("array(double)") Block first, @SqlType(
         return dotProduct;
     }
 
+    @Description("Calculates the cosine similarity between two vectors")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cosineSimilarity(@SqlType("array(double)") Block first, @SqlType("array(double)") Block second)
+    {
+        return 1.0 - cosineDistance(first, second);
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/scalar/ArrayVectorFunctions.java:64]: I'd do it the other way around. Implement this method using the logic in cosineDistance (except for the bit at the end), and make cosineDistance = 1 - cosineSimilarity.

**Review by martint — APPROVED** [review]

**Comment by ebyhr** [comment]
Why is this PR merged without tests?

**Comment by mosabua** [comment]
Is this a breaking change since the older function is renamed? Also .. don't we need to update docs?

**Comment by martint** [comment]
No, it’s a new function.

**Comment by mosabua** [comment]
Yeah .. sorry ... I misread the diff .. I will send a PR for the docs addition.

**Comment by mosabua** [comment]
Hm.. got questions in the docs PR..

**Pull request merged**